window_manager/src/XWindow.cpp

```diff
@@ -73,14 +75,14 @@ XWindow::XWindow(WindowConfig &&config, Display *root_display, int screen_number
         }
             break;
     }
-    XSetStandardProperties(rendering_context_->x_display, rendering_context_->gl_x_window, config_.caption.c_str(), "",None,nullptr,0,nullptr);
+    XSetStandardProperties(rendering_context_->get_display(), rendering_context_->get_drawable(), config_.caption.c_str(), "",None,nullptr,0,nullptr);
 
     if (config_.process_input) {
         if (config.editor) {
-            input_manager_.emplace(EditorInputManager(rendering_context_));
+            input_manager_ = std::make_shared<AbstractInputManager>(EditorInputManager(EditorInputManager::handler_collection{}));
```

Comment:
It so looks as window has no need to create the input manager itself, add a TODO, to remove this later.